perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java

```diff
@@ -137,6 +144,70 @@ public static String join(Iterable<?> collection, String separator) {
 		return oBuilder.toString();
 	}
 
+	/**
+	 * Returns additionalUserExtSources from the subject. It's used for synchronization from different ExtSources. subjectFromLoginSource was obtained from the login ExtSource.
```

Comment:
Are you sure it must be subject from login source? I think it can be subject from any source.